xdsl/transforms/experimental/live_range.py

```diff
@@ -0,0 +1,52 @@
+from dataclasses import dataclass
+from xdsl.ir import Block, OpResult, SSAValue
+from xdsl.utils.exceptions import InvalidIRException
+
+
+@dataclass(init=False, frozen=True)
+class LiveRange:
+    """
+    Represents the live range of an SSA value as half-open interval using the
+    index of an Operation within its parent Block.
+    """
+
+    value: SSAValue
+    start: int
+    end: int
+
+    def __init__(
+        self, value: SSAValue, start: int | None = None, end: int | None = None
+    ) -> None:
+        if start is not None and end is not None:
+            object.__setattr__(self, "value", value)
```

Comment:
? why do you need this init

Reply:
If you want to pass manually computed live-ranges to test the linear scan algorithm without relying on the automatically computed live-ranges that's kinda necessary.

Reply:
Sorry what I mean is the auto generated init would avoid this, I'd recommend doing further verification in `__post_init__(self)`